js/index.js

```diff
@@ -1,7 +1,113 @@
-// Iteration 1: Names and Input
 
+// // iteration 1
 
-// Iteration 2: Conditionals
+let hacker1 = "Angel"
+console.log ( `The driver's name is ${hacker1}`)
+let hacker2 = "Shirley"
+console.log (`The navigator's name is ${hacker2}`)
+
+// // iteration 2
+
+if (hacker1.length > hacker2.length) {
+    console.log(`The driver has the longest name, it has ${hacker1.length} characters`)
+} else if (hacker1.length < hacker2.length) {
+    console.log (`It seems that the navigator has the longest name, it has ${hacker2.length} characters`)
+} else {
+    console.log (`Wow, you both have equally long names, ${hacker2.length}  characters!`)
+}
+
+// iteration 3.1
+
+let name = hacker1.toUpperCase().split('').join(' ')
+console.log(name)
+
+// iteration 3.2
+
+let nameReverse = hacker1.split('').reverse().join('')
+console.log(nameReverse)
```

Comment:
Aunque esto es correcto, en el ejercicio se pedía no utilizar esta clase de propiedades, se quería que se utilizasen los bucles `for` or `while` o `do while` y con los condicionales `if else`

Reply:
Esos métodos los encontramos en Google y creo que los entendimos. Según las sugerencias nos hemos creado nuestras propias funciones con for que reemplazan los métodos que hemos utilizado.